### PR TITLE
Nd 386 3 - creating secrets in secrets manager, task definitions sourcing secrets from secrets manager

### PR DESCRIPTION
**Pull request by asifamirat00**

### CODE REVIEW

File: modules/admin/iam.tf [hunk 1]
```diff
@@ -60,6 +60,37 @@ data "aws_iam_policy_document" "assume_role_policy" {
...
+        "Effect" : "Allow",
+        "Action" : [
+          "secretsmanager:GetResourcePolicy",
+          "secretsmanager:GetSecretValue",
+          "secretsmanager:DescribeSecret",
+          "secretsmanager:ListSecretVersionIds",
+          "secretsmanager:ListSecrets"
+        ],
+        "Resource" : [
+          "*"
```
Comment (smjmoj) [inline modules/admin/iam.tf:82]: Should this * (all) be more targeted to the secrets' initial path ?
`
"Resource" : [
          "*:${data.aws_caller_identity.target_account.account_id}:parameter/codebuild/dhcp/*"
          `

File: modules/dns_dhcp_common/iam.tf [hunk 2]
```diff
@@ -83,6 +83,37 @@ resource "aws_iam_role_policy" "ecs_task_policy" {
...
+        "Action" : [
+          "secretsmanager:GetResourcePolicy",
+          "secretsmanager:GetSecretValue",
+          "secretsmanager:DescribeSecret",
+          "secretsmanager:ListSecretVersionIds",
+          "secretsmanager:ListSecrets"
+        ],
+        "Resource" : [
+          "*"
+        ]
```
Comment (smjmoj) [inline modules/dns_dhcp_common/iam.tf:106]: Should this * (all) be more targeted to the secrets' initial path ?
`
"Resource" : [
          "*:${data.aws_caller_identity.target_account.account_id}:parameter/codebuild/dhcp/*"
          `

File: modules/admin/iam.tf (hunk 1 above)
Comment (jamesgreen-moj) [reply modules/admin/iam.tf:82]: I do not believe the suggestion is possible due to the resource requiring a secret arn and not an SSM parameter arn.  I could lock this down a bit more if I do something like this:
`        "Resource" : [
          values(var.secret_arns)
        ]
`

Then pass in the secret_arns from the secrets_manager.tf

Comment (jamesgreen-moj) [reply modules/admin/iam.tf]: @smjmoj I have now updated to lockdown the policy more: https://github.com/ministryofjustice/staff-device-dns-dhcp-infrastructure/pull/341/commits/71d53067fdcce45e587fb89e7c7a5b6bc2fabc17

**Review by smjmoj — APPROVED** [review]
LGTM

**Pull request merged**